crates/iceberg/src/spec/schema.rs

```diff
@@ -705,6 +734,36 @@ mod tests {
     use crate::spec::schema::_serde::SchemaV2;
     use std::collections::HashMap;
 
+    fn check_schema_serde(json: &str, expected_type: Schema) {
+        let desered_type: Schema = serde_json::from_str(json).unwrap();
+        assert_eq!(desered_type, expected_type);
+
+        let sered_json = serde_json::to_string(&expected_type).unwrap();
+        assert_eq!(sered_json, json);
+        let parsed_json_value = serde_json::from_str::<Schema>(&sered_json).unwrap();
+
+        assert_eq!(parsed_json_value, desered_type);
+    }
+
+    #[test]
+    fn test_serde() {
```

Comment:
Could we also add some test for v1 without `schema-id`?

Reply:
Done, I implemented a test with schema-id set and with schema-id unset. I also added a check to be sure that V1 or V2 enum is correctly chosen depending if schema-id is set or unset.
I change the test schema to rely on the existing table_simple_schema already existing instead of relying on other one
I also tied the json string to table_simple_schema so that if we change the function, we also know we need to change the json string accordingly